src/main/java/org/example/NetworkUtils.java

```diff
@@ -35,4 +35,12 @@ public static int getNamesrvPort() {
         }
         return Integer.parseInt(port);
     }
+
+    public static String getServerIp() {
+        String ip = System.getenv("rocketmq.proxy.ip");
```

Comment:
When will we set `rocketmq.proxy.ip` here?

Reply:
This will be used when the IP address your container is reachable on is something other then `127.0.0.1`. Specifically, the two scenarios I've seen this happen with are rootless Podman and a remote Docker TCP bind.

Reply:
> This will be used when the IP address your container is reachable on is something other then `127.0.0.1`. Specifically, the two scenarios I've seen this happen with are rootless Podman and a remote Docker TCP bind.

I have not seen `rocketmq.proxy.ip` was set in opentelemetry instrumentation, so you will set it manually?

Reply:
OK, I found it in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11222